include/dd/Package.hpp

```diff
@@ -508,8 +508,6 @@ namespace dd {
                             } else { // pos. control
                                 em[i] = makeDDNode(z, std::array{(i1 == i2) ? makeIdent(static_cast<Qubit>(start), static_cast<Qubit>(z - 1)) : mEdge::zero, mEdge::zero, mEdge::zero, em[i]});
                             }
-                        } else { // not connected
```

Comment:
It could be worth to introduce a separate routine that handles the no-control case as this becomes almost trivial now. Technically one could use the routine that takes a matrix and creates a DD out of that.

The function here could then become something like
```c++
if (controls.empty()) {
 return ... [call special function for just constructing the single qubit operation matrix] ...
}

... [do the usual] ...
```

Reply:
I added an if statement to check if there are no controls as you said. This allows us to avoid the for loops over the levels of the DD for the control construction if there are none. No need for an extra function.

This issue is probably resolved.